added starter doc (#75)
dappforce/subsocial-docs: docs/develop/SubsocialStarter.md

---
id: subsocial-starter
title: Quick Start
displayed_sidebar: developSidebar
---

> ### Get started building on Subsocial quickly using Subsocial Starter

Reviewer: ```suggestion
### Get started building on Subsocial quickly using the Subsocial Starter
```

> **Subsocial Starter** is a create-react-app template that gives you initial template to start building on Subsocial. It has all the necessary dependencies and configurations that are required to quickly start.

Reviewer: ```suggestion
**Subsocial Starter** is a create-react-app template that gives you an initial template to start building on Subsocial. It has all of the necessary dependencies and configurations that are required to quickly start.
```

:::info

> Starter is the foundational layer for using Subsocial JS SDK in a React app.

Reviewer: ```suggestion
The Starter is the foundational layer for using the Subsocial JS SDK in a React app.
```

:::

> The starter template provides the following features:

Reviewer: ```suggestion
The Starter template provides the following features:
```

- Installed package dependencies like **@subsocial/api** and **@polkadot/api**
- API Configurations for **Testnet** and **Mainnet** to switch easily
- React context hooks for Subsocial APIs automatically connects and exposes necessary methods
- Polkadotjs Wallet methods to easily fetch connected accounts and send transactions for signing

> ### Create new project

Reviewer: ```suggestion
### Create a new project
```

> Open terminal, and run the following command:

Reviewer: ```suggestion
Open the terminal, and run the following command:
```

```typescript
  npx create-react-app my-app --template subsocial-starter
```

> Now, the project is created inside `/my-app` directory.

Reviewer: ```suggestion
Now, the project is created inside the `/my-app` directory.
```

> Run the following command next in the terminal:

Reviewer: ```suggestion
Next, run the following command in the terminal:
```

```typescript
  cd my-app && yarn
```

> Make sure you use `yarn` to install all dependencies, npm might not work.

> Update `package.json` file with project details. 

Reviewer: ```suggestion
Update `package.json` file with the project details. 
```

> ### Running new project

Reviewer: ```suggestion
### Running a new project
```

> To run new project, just do:

Reviewer: ```suggestion
To run a new project, just do:
```

```typescript
  yarn start
```

> ### About Project Structure

Reviewer: ```suggestion
### About the project structure
```

Understanding the project structure:

> - The **config.ts** file contains the objects to connect to different networks like Subsocial Testnet and Mainnet. 

Reviewer: ```suggestion
- The **config.ts** file contains the objects to connect to different networks like the Subsocial Testnet and Mainnet. 
```

- The **provider.tsx** file contains the context provider & hook for Subsocial related methods like:
```js
export default { isReady, api, initialize, network, changeNetwork, setupCrustIPFS }
```

> - The **polkadotjs.ts** file exposes methods for connecting and getting data from polkadot js wallet extension.

Reviewer: ```suggestion
- The **polkadotjs.ts** file exposes methods for connecting and getting data from the polkadot.js wallet extension.
```

```js
export default { logTransaction, getAllAccounts, signAndSendTx }
```

![Project Structure](../../static/img/starter-structure.png)

> ### How build using the Starter?

Reviewer: ```suggestion
### How to build using the Starter
```

> Once the project is setup, you can go through the **App.tsx** file that contains the basic usage of connecting to testnet, fetching a space and creating a new space.

Reviewer: ```suggestion
Once the project is setup, you can go through the **App.tsx** file that contains the basic usage of connecting to the testnet, fetching a space, and creating a new space.
```

Now, you can start exploring the ideas and build on Subsocial.

> We have list of methods to read and do transactions on chain. Try out the code snippets in [**Subsocial Playground**](https://play.subsocial.network)

Reviewer: ```suggestion
We have list of methods to read and do transactions on chain. Try out the code snippets in the [**Subsocial Playground**](https://play.subsocial.network)
```

The following pointers to have in mind: 

> - Build on **Testnet** first. And get some test tokens [here](/docs/develop/getting-started/testnet)

Reviewer: ```suggestion
- Build on the **Testnet** first. And get some test tokens [here](/docs/develop/getting-started/testnet)
```

> - Before experimenting locally, try the code snippets on [Playground](https://play.subsocial.network)

Reviewer: ```suggestion
- Before experimenting locally, try the code snippets on the [Playground](https://play.subsocial.network)
```

> - Go through [bounties](https://github.com/dappforce/subsocial-bounties) repository, to get some project interesting project ideas

Reviewer: ```suggestion
- Go through the [bounties](https://github.com/dappforce/subsocial-bounties) repository to get some interesting project ideas
```

> - Facing issues? Report on [Discord](https://discord.gg/HWzHdps7) or [Telegram](https://t.me/+ZzvLu0ZfkQwxNGQy)

Reviewer: ```suggestion
- Facing issues? Report them in our [Discord](https://discord.gg/HWzHdps7) or [Telegram](https://t.me/+ZzvLu0ZfkQwxNGQy) channels
```
